ledger-service/db-backend/src/main/scala/com/digitalasset/http/dbbackend/Queries.scala

```diff
@@ -782,8 +792,13 @@ private final class OracleQueries(tablePrefix: String) extends Queries(tablePref
     sql"""CREATE INDEX $stakeholdersIndexName ON $contractStakeholdersViewName (tpid, stakeholder)"""
   )
 
+  private[this] val contractKeyHashIndexName = Fragment.const0(s"${tablePrefix}ckeyHash_idx")
+  private[this] def contractKeyHashIndex = CreateIndex(
```

Comment:
Same niptick as before.

Reply:
Likewise, we only refer to the definition once, so no work is saved by making it a `val`, and reordering/refactoring may introduce initialization bugs. Better to keep it a `def`.

Reply:
Ok, I thought there were no problems in having a `val` in an `abstract class`.

Reply:
```scala
scala> abstract class Foo{def foo:String; val bar = s"$foo hi"}; class Bar extends Foo{val foo = "foo"}
class Foo
class Bar

scala> (new Bar).bar
val res2: String = null hi
```